spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jNativeTypesFailureAnalyzer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.data.neo4j;
+
+import org.neo4j.ogm.driver.NativeTypesException;
+import org.neo4j.ogm.driver.NativeTypesNotAvailableException;
+import org.neo4j.ogm.driver.NativeTypesNotSupportedException;
+
+import org.springframework.boot.diagnostics.AbstractFailureAnalyzer;
+import org.springframework.boot.diagnostics.FailureAnalysis;
+
+/**
+ * A {@link org.springframework.boot.diagnostics.FailureAnalyzer} that performs analysis
+ * of a Neo4j-OGM configuration failure caused by a missing native types module.
+ *
+ * @author Michael Simons
+ */
+class Neo4jNativeTypesFailureAnalyzer
```

Comment:
I am not entirely convinced that's worth adding a dedicated `FailureAnalyzer`. You should be able to phrase that in the exception message itself, can't you?

Reply:
I could but than people would have to find it in the stack trace. I like the value provided by a `FailureAnalyzer` for providing clear instructions which can be seen much better than in a stack trace.

Reply:
I am not denying any of that. But I don't see why those two particular case should have a dedicated `FailureAnalyzer`. Have you considered using `InvalidConfigurationPropertyValueException`?

Reply:
Oh nice, didn't know that. I'll have a look.